sklearn/compose/_column_transformer.py

```diff
@@ -690,7 +783,7 @@ def _get_transformer_list(estimators):
     columns = [trans[0] for trans in estimators]
     names = [trans[0] for trans in _name_estimators(transformers)]
```

Comment:
I just realized if we want to stay strictly barckward compatible, we should use the same `_guess_order` mechanism here, otherwise the `names` are gonna be taken from the columns instead of transformers (I guess we should have a test to test this as well). Meaning I need to take those functions out of the class, and use it here as well.

This is getting dirtier than what I like. I'd love to see if it can be done cleaner.